syntax = "proto3";

import "google/protobuf/descriptor.proto";

message CmdOptions {  
	TS_CMD ts_cmd = 1;
    CS_CMD cs_cmd = 2;
    SC_CMD sc_cmd = 3;
}

extend google.protobuf.MessageOptions {
    CmdOptions cmd = 1011;
}

enum TS_CMD
{
	CMD_BEGIN  = 0;
	CMD_SC_KEY = 1; // key下发
	CMD_PING   = 2; // pingpong校验对
	CMD_PONG   = 3;

	CMD_WORLD_BEGIN = 10;
	CMD_WORLD_END	= 999;

	CMD_ZONE_BEGIN = 1000;
	CMD_ZONE_END   = 3999;

	CMD_INTERRUPT = 9999;
	CMD_MAX = 10000;
};


enum CS_CMD
{
	CS_CMD_NONE  = 0;
	CMD_CS_LOGIN	   = 10; //登录
	CMD_CS_CREATEACTOR = 11; //创建角色
	CMD_CS_SELECTACTOR = 12; //选择角色进入游戏


	CMD_CS_LOADMAP_SUCC = 1001; //加载地图完成
	CMD_CS_MOVE			= 1002; //通知移动
	CMD_CS_CHANGEMAP	= 1003;

	CMD_CS_CASTSKILL		  = 1004; //通知释放技能
	CMD_CS_SKILL_BREAK		  = 1005; //通知打断技能
	CMD_CS_SKILL_CHGTARGETPOS = 1006; //通知服务器切换目标点
	CMD_CS_CHANGE_PKMODE	  = 1007;
	CMD_CS_REBORN			  = 1008; //请求复活
	


	CMD_CS_ITEM_SWAP	= 1020; //物品交换
	CMD_CS_ITEM_SPLIT	= 1021; //物品分离
	CMD_CS_ITEM_COMBINE = 1022; //物品合并
	CMD_CS_ITEM_TIDY	= 1023; //整理背包
	CMD_CS_ITEM_USE		= 1024; //使用物品

	CMD_CS_ITEM_STORAGE_OPEN	 = 1025; //打开仓库
	CMD_CS_ITEM_STORAGE_CHECKIN	 = 1026; //存入仓库
	CMD_CS_ITEM_STORAGE_CHECKOUT = 1027; //从仓库取出

	CMD_CS_EQUIP   = 1030; //穿装备
	CMD_CS_UNEQUIP = 1031; //脱装备



	CMD_CS_TALK		   = 1040; //聊天信息
	CMD_CS_TASK_SUBMIT = 1041; //手动提交任务

	CMD_CS_ACTIVE_NPC	= 1042; //点击NPC
	CMD_CS_DIALOG_CLICK = 1043; //点击对话框按钮
	CMD_CS_ACHI_TAKE	= 1044; //成就领取




	CMD_CS_TEAMCREATE		= 1050; //创建队伍
	CMD_CS_TEAMQUIT			= 1051; //退出队伍
	CMD_CS_TEAMKICKMEMBER	= 1052; //踢出队员
	CMD_CS_TEAMINVITEMEMBER = 1053; //邀请队员

	CMD_CS_TEAMACCEPTINVITE = 1054; //同意邀请
	CMD_CS_TEAMNEWLEADER	= 1055; //提升为队长
	CMD_CS_TEAMAPPLYMEMBER	= 1056; //申请加入
	CMD_CS_TEAMACCEPTAPPLY	= 1057; //同意申请


	CMD_CS_GUILDCREATE			= 1070; //创建公会
	CMD_CS_GUILDQUIT			= 1071; //退出公会
	CMD_CS_GUILDKICKMEMBER		= 1072; //踢出公会
	CMD_CS_GUILDINVITEMEMBER	= 1073; //邀请入会
	CMD_CS_GUILDACCEPTINVITE	= 1074; //同意邀请
	CMD_CS_GUILDNEWLEADER		= 1075; //新会长
	CMD_CS_GUILDAPPLYMEMBER		= 1076; //申请入会
	CMD_CS_GUILDACCEPTAPPLY		= 1077; //同意申请
};

enum SC_CMD
{
	SC_CMD_NONE	 = 0;
	CMD_SC_LOGIN		 = 10; //登录结果
	CMD_SC_ACTORINFOLIST = 11; //角色列表
	CMD_SC_WAITINFO		 = 12; //进入等待队列
	CMD_SC_CREATEACTOR	 = 13; //创建角色结果
	CMD_SC_SELECTACTOR	 = 14; //选择角色进入游戏结果

	CMD_SC_LOADMAP		 = 1001; //通知加载地图
	CMD_SC_ENTERMAP		 = 1002; //通知进入地图
	CMD_SC_PLAYERINFO	 = 1003; //主角信息
	CMD_SC_ACTORATTRIB	 = 1004; //主角最终属性
	CMD_SC_COMMONDATA	 = 1005; //主角公共数据
	CMD_SC_DATACOUNT	 = 1006; //主角计数器数据
	CMD_SC_SYSVAR		 = 1007; //系统变量通知
	CMD_SC_COOLDOWN		 = 1008; //冷却器数据
	CMD_SC_MAPVAL		 = 1009; //sync mapval

	CMD_SC_AOI_NEW		 = 1020; //广播建立对象
	CMD_SC_AOI_REMOVE	 = 1021; //广播删除对象
	CMD_SC_AOI_UPDATE	 = 1022; //广播移动

	CMD_SC_CASTSKILL	 = 1031; //广播技能释放
	CMD_SC_SKILL_BREAK	 = 1032; //广播技能打断
	CMD_SC_SKILL_DAMAGE	 = 1033; //自己收到技能伤害列表
	CMD_SC_SKILL_STUN	 = 1034; //广播技能进入僵直阶段
	CMD_SC_SKILL_EFFACT	 = 1035; //广播技能伤害1次
	CMD_SC_DAMAGE		 = 1036; //通知自己被伤害
	CMD_SC_DEAD			 = 1037; //广播死亡
	CMD_SC_ATTRIB_CHANGE = 1038; //属性变化

	CMD_SC_CHANGE_PKMODE = 1040;
	CMD_SC_STATUS_LIST	 = 1041; //状态列表
	CMD_SC_STATUS_INFO	 = 1043; //状态数据
	CMD_SC_STATUS_ACTION = 1044; //状态作用1次

	CMD_SC_ITEM_DELETE = 1050; //发送物品信息
	CMD_SC_ITEM_INFO   = 1051; //发送物品信息
	CMD_SC_ITEM_CHANGE = 1052; //发送物品信息改变
	CMD_SC_TALK		   = 1053; //聊天信息

	CMD_SC_TASK_INFO = 1060; //任务信息
	CMD_SC_TASK_DATA = 1061; //任务数据变化
	CMD_SC_DIALOG	 = 1062; //打开对话框
	CMD_SC_ACHI_INFO = 1063; //成就达成情况

	CMD_SC_TEAMMEMBER_INFO	 = 1070; //队员信息
	CMD_SC_TEAMMEMBER_ACTION = 1071; //成员加入/离开/被kick
	CMD_SC_TEAMINVITEMEMBER	 = 1072; //邀请队员
	CMD_SC_TEAMAPPLYMEMBER	 = 1073; //申请加入

	CMD_SC_GUILDINVITEMEMBER	= 1080; //邀请入会
	CMD_SC_GUILDMEMBER_INFO		= 1081; //成员信息
	CMD_SC_GUILDMEMBER_ACTION	= 1082;	//操作
	CMD_SC_GUILDAPPLYMEMBER		= 1083;	//申请入会
};


message SC_KEY
{
	option (cmd).ts_cmd = CMD_SC_KEY;
	uint32 key = 1;
};
